### PR TITLE
GP Unit accounts migrated as BC Statistical accounts

### LINKED ISSUE #26487: [INTERNAL] GP Updates
### Describe the issue

Issue to link GP update PRs to.

### Expected behavior

GP Updates merged.

### Steps to reproduce

N/A

### Additional context

_No response_

### I will provide a fix for a bug

- [X] I will provide a fix for a bug

### PR DESCRIPTION
**Pull request by jaymckinney-enavate**
This PR enhances the GP migration by migrating Unit accounts as BC Statistical accounts.

- Updated the Account migrator to include Unit accounts, and handle them separately from the G/L Account related code.
- Statistical Account posting logic.
- Added count of unposted Statistical account batches to the Failed Batches tile.
- Added unit tests.

Fixes #26487
Fixes [AB#537203](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/537203)

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al [hunk 1]
```diff
@@ -232,36 +335,68 @@ codeunit 4017 "GP Account Migrator"
...
+        AccountNum: Code[20];
+    begin
+        AccountNum := CopyStr(GPAccount.AcctNum.Trim(), 1, MaxStrLen(StatisticalAccount."No."));
+
+        if StatisticalAccount.Get(AccountNum) then
+            exit;
+
+        DataMigrationErrorLogging.SetLastRecordUnderProcessing(Format(GPAccount.RecordId));
+
+        StatisticalAccount.Init();
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:351]: Minor - Init is useless, everything is initialized. It is used only to clear the non-primary key fields. Recommendaiton is to delete

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al [hunk 2]
```diff
@@ -232,36 +335,68 @@ codeunit 4017 "GP Account Migrator"
...
+        StatisticalAccount."No." := AccountNum;
+        StatisticalAccount.Name := CopyStr(GPAccount.Name, 1, MaxStrLen(StatisticalAccount.Name));
+        StatisticalAccount.Insert(true);
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:354]: I would recommend to set the global dimension 1 and 2 Code from the segments.

**Review by nikolakukrika — APPROVED** [review]

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al (hunk 1 above)
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al]: Updates, thanks

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al (hunk 2 above)
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:361]: Updates, thanks

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al [hunk 3]
```diff
@@ -212,7 +222,100 @@ codeunit 4017 "GP Account Migrator"
...
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        GPFiscalPeriods: Record "GP Fiscal Periods";
+        StatisticalAccJournalBatch: Record "Statistical Acc. Journal Batch";
+        StatisticalAccJournalLine: Record "Statistical Acc. Journal Line";
+        StatisticalAccJournalLineCurrent: Record "Statistical Acc. Journal Line";
+        InitialYear: Integer;
+        BeginningBalance: Decimal;
+        DocumentNo: Code[20];
+        LineNum: Integer;
+        ACTNUMBR_1: Code[20];
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:239]: Minor, we could use a list here

**Review by nikolakukrika — APPROVED** [review]

File: Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al (hunk 3 above)
Comment (JesperSchulz) [reply Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al:239]: We'll take it "as is".

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 20391.

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 24.

Build ID to track: <None>.

**Pull request merged**